gateway-messages/src/sp_impl.rs

```diff
@@ -350,10 +351,18 @@ pub trait SpHandler {
         value: &[u8],
     ) -> Result<(), SpError>;
 
-    fn get_caboose_value(
+    fn get_component_caboose_value(
         &mut self,
+        component: SpComponent,
+        slot: u16,
         key: [u8; 4],
-    ) -> Result<&'static [u8], SpError>;
+    ) -> Result<CabooseValue, SpError>;
+
+    fn copy_caboose_value_into(
+        &self,
+        value: CabooseValue,
+        out: &mut [u8],
```

Comment:
I assume the implementer can assume `out.len()` matches what it is described by `value.len()`? Would it ever be possible that `value.len()` describes a max length, but we don't know the exact length until we get into this method (and therefore we should return a `usize` of the number of bytes written to `out`)?

Reply:
Hmmm, this suggests that `copy_caboose_value_into` should take a bare `addr: u32` instead of a `value: CabooseValue`, to avoid this ambiguity.

Reply:
Ah, that won't work, because `CabooseValue` is an `enum`, not a simple position + length.  I'll add some commentary here to explain the invariants.